build/net/src/lib.rs

```diff
@@ -45,7 +58,35 @@ pub struct TaskNote {
 }
 
 pub fn load_net_config() -> Result<NetConfig, Box<dyn std::error::Error>> {
-    Ok(build_util::config::<GlobalConfig>()?.net)
+    let cfg = build_util::config::<GlobalConfig>()?.net;
+
+    match (cfg!(feature = "vlan"), cfg.vlan.is_some()) {
```

Comment:
Are you sure this check is working? I thought you had to detect features from build scripts using the `CARGO_FEATURE_VLAN` (in this case) environment variable.